src/main/jhipster/src/main/webapp/app/study-manager/advance/abstract-advance.component.ts

```diff
@@ -282,6 +345,129 @@ export abstract class AbstractAdvanceComponent implements OnInit {
         }
     }
 
+    // methods for advance preview
+    resetTable(): void {
+        this.page = 1;
+        this.previousPage = 1;
+        this.completePreviewList = [];
+        this.listPerPage = [];
+    }
+
+    onSuccess(data: AdvancedGermplasmPreview[], fromDelete= false) {
+        this.completePreviewList = data;
+        this.processPagination(this.completePreviewList);
+        this.loadPage(1, fromDelete);
+        this.isPreview = true;
+    }
+
+    loadPage(page: number, forceReload = false) {
+        if (page !== this.previousPage || forceReload) {
+            this.previousPage = page;
+            this.currentPagePreviewList = this.listPerPage[page - 1];
+            const itemCount = this.currentPagePreviewList.length;
+            this.currentPageCount = ((page - 1) * this.itemsPerPage) + itemCount;
+        }
+    }
+
+    exitPreview() {
+        this.isPreview = false;
+    }
+
+    applyFilters() {
+        this.page = 1;
+        this.previousPage = 1;
+        const filteredList = this.completePreviewList.filter(
+            (row) => {
+                const env = (row.trialInstance + '-' + row.locationName).toLowerCase();
+                if (this.filters.environment.value && !env.includes(this.filters.environment.value.toLowerCase())) {
+                    return false;
+                }
+
+                if (this.filters.plotNumber.value && row.plotNumber !== this.filters.plotNumber.value) {
+                    return false;
+                }
+
+                if (this.filters.plantNumber.value && row.plantNumber !== this.filters.plantNumber.value) {
+                    return false;
+                }
+
+                if (this.filters.entryNumber.value && row.entryNumber !== this.filters.entryNumber.value) {
+                    return false;
+                }
+
+                if (this.filters.cross.value && !row.cross.toLowerCase().includes(this.filters.cross.value.toLowerCase())) {
+                    return false;
+                }
+
+                if (this.filters.immediateSource.value && !row.immediateSource.toLowerCase().includes(this.filters.immediateSource.value.toLowerCase())) {
+                    return false;
+                }
+
+                if (this.filters.breedingMethod.value && !row.breedingMethodAbbr.toLowerCase().includes(this.filters.breedingMethod.value.toLowerCase())) {
+                    return false;
+                }
+
+                return true;
+            }
+        );
+
+        this.processPagination(filteredList);
+        this.loadPage(1, true);
+    }
+
+    processPagination(list: AdvancedGermplasmPreview[]) {
+        this.totalItems = list.length;
```

Comment:
Please check the total items in the pagination section. It should be the size of all items in the list.
<img width="802" alt="Screenshot 2023-03-20 at 12 30 09 PM" src="https://user-images.githubusercontent.com/5011207/226247439-48c1da7d-80ec-4068-8e7d-7265a84dae59.png">